### ISSUE TITLE
Does geofencing require "Always" location permission?

### ISSUE THREAD

**Issue by joserocha3** [transistorsoft/flutter_background_geolocation#95]
## Your Environment
* Plugin version: 1.0.10
* Platform: iOS
* OS version: 12.2
* Device manufacturer / model: Simulator and iPad
* Flutter info (`flutter doctor`):
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.5.4-hotfix.2, on Mac OS X 10.14.5 18F132, locale en-US)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.2.1)
[✓] Android Studio (version 3.3)
[✓] VS Code (version 1.32.3)
[✓] Connected device (2 available)

• No issues found!
```
* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
 bg.Config(
      desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
      persistMode: bg.Config.PERSIST_MODE_GEOFENCE,
      geofenceProximityRadius: 100000, // about 62 miles
      logLevel: bg.Config.LOG_LEVEL_ERROR,
      stopOnTerminate: false,
      enableHeadless: true,
      startOnBoot: true,
      autoSync: true,
      reset: _reset ?? true,
      // Do not record location event if more than 300 meters/second since last
      // recorded event - Android only, iOS does not allow mock locations on device
      speedJumpFilter: 300,
      foregroundService: true,
      disableLocationAuthorizationAlert: true,
      notification: bg.Notification(
        title: 'Zone Tracker',
        text: 'Search underway',
        channelName: 'Zone Tracker',
        color:
            '#${Constants.colors.primary.value.toRadixString(16).substring(2)}',
        smallIcon: 'drawable/ic_notification',
        largeIcon: 'drawable/ic_notification',
        priority: bg.Config.NOTIFICATION_PRIORITY_MIN,
      ),
    );
```

## Expected Behavior
NA 

## Actual Behavior
NA

## Steps to Reproduce
NA

## Context
This is more of a question. If we do not request "Always" location permission, or user selects "When in use" will geofencing still work? 

Asking specifically for iOS because of app being **rejected**. Wondering if it works before we start refactoring our app to test it out.

Granting "Always" works flawlessly including when device does not have a data connection.

## Debug logs
NA

**Comment by christocracy** [comment]
Always is required for monitoring geofences, yes.

**Comment by joserocha3** [comment]
Hi @christocracy, looks like Apple is saying `Always` is not needed for geofencing (Monitoring Geographical Regions). Is there something extra that the plugin does that requires `Always`?

I am trying to figure out how to reply to Apple so they understand why it is required.

Thanks!

**Comment by joserocha3** [comment]
I know you calculate closest geofences every so often in order to _refresh_ the 20 being tracked. Perhaps I can mention that. What do you think?

**Comment by christocracy** [comment]
This has nothing to do with the plugin’s geospatial querying of geofences.

If you want a geofence to be able to fire “always” (ie: background / terminated), you need Always authorization.

If you only need a geofence to fire when your app is in the foreground (pretty useless), you would be happy with “When In Use” authorization.

These terms Always vs WhenInUse *really do* mean what they say.

**Comment by joserocha3** [comment]
Got it. Thanks!

**Comment by joserocha3** [comment]
@christocracy got them to approve it 👍

**Comment by danPudwellUK** [comment]
I know its been a couple of years but having the same issue with Apple.
@joserocha3 do you know what you said to Apple to get them to approve it?

**Comment by joserocha3** [comment]
@danPudwellUK hmmm... I don't recall exactly and I can't find any emails about it. From what I remember I think we just had to be pretty descriptive in the permission descriptions found in the plist.

**Comment by danPudwellUK** [comment]
oddly they have now approved it after 4 weeks of arguing and no changes made to the app

**Comment by joserocha3** [comment]
Oh geez. Well there should be less pushback about location permissions when submitting updates now that it was initially approved.
